File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math

def solution(n, words):
    dict={}
    for index,word in enumerate(words):
        num=(index+1)%n
        if num == 0:
            num = n
        turn= math.ceil((index+1)/n)
        # 끝말잇기 탈락
        if index>0 and words[index-1][-1] !=word[0]:
            return [num,turn]
        # 중복 탈락
        if word in dict.keys():
            return [num,turn]
        else :
            dict[word]=True

    return [0,0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let answer = [];
    let wordSet = new Set();
    let count = 0;
    let isFail = false;
    let lastChar = null;
    
    for(const word of words){
        if(lastChar !== null){
            if(wordSet.has(word) || lastChar !== word[0]){
            isFail=true;
            break;
            }
        }
        
        wordSet.add(word);
        lastChar = word.at(-1);
        count += 1;
    }
    
    if(isFail){
        answer.push(count % n +1);
        answer.push(Math.floor(count/n) +1);
    } else{
        return [0,0];
    }
    

    return answer;
}